### PR TITLE
Implement additional GPU validation fixes

### PR DESCRIPTION
**Pull request by eddie-c-davis**

### CODE REVIEW

File: fv3core/stencils/a2b_ord4.py [hunk 1]
```diff
@@ -290,8 +290,7 @@ def extrapolate_corner_qout(qin, qout, i, j, kstart, nk, corner):
         qin[i + i3a, j + j3a, kslice],
         qin[i + i3b, j + j3b, kslice],
     )
-    r3 = 1.0 / 3.0
-    qout[i, j, kslice] = (ec1 + ec2 + ec3) * r3
+    qout[i, j, kslice] = (ec1 + ec2 + ec3) * (1.0 / 3.0)
```
Comment (rheacangeo) [inline fv3core/stencils/a2b_ord4.py:293]: That's great this works. It was written this way to get better validation matching the fortran (as it does changing the numbers ever so slightly). But if this is passing the tests, they A+.

File: fv3core/stencils/remap_profile.py [hunk 2]
```diff
@@ -297,23 +294,17 @@ def set_inner_as_kord9(
...
-        elif extm and extm[0, 0, 1]:
-            a4_2 = a4_1
-            a4_3 = a4_1
-            a4_4 = 0.0
-        elif extm > 0.0 and (qmin > 0.0 and a4_1 < qmin):
+        if (
+            (extm != 0.0 and extm[0, 0, -1] != 0.0)
+            or (extm != 0.0 and extm[0, 0, 1] != 0.0)
+            or (extm > 0.0 and (qmin > 0.0 and a4_1 < qmin))
+        ):
```
Comment (rheacangeo) [inline fv3core/stencils/remap_profile.py:301]: sweet!

File: fv3core/stencils/remap_profile.py [hunk 3]
```diff
@@ -297,23 +294,17 @@ def set_inner_as_kord9(
...
             a4_2 = a4_1
             a4_3 = a4_1
             a4_4 = 0.0
         else:
-            diff_23 = a4_2 - a4_3
             a4_4 = 6.0 * a4_1 - 3.0 * (a4_2 + a4_3)
-            if abs(a4_4) > abs(diff_23):
+            if abs(a4_4) > abs(a4_2 - a4_3):
```
Comment (rheacangeo) [inline fv3core/stencils/remap_profile.py:307]: I'm so glad we can do this now

File: tests/test_translate.py [hunk 4]
```diff
@@ -18,6 +18,9 @@
 
 OUTDIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "output")
 GPU_MAX_ERR = 1e-10
+GPU_NEAR_ZERO = 1e-15
+
+_near_zero = 1e-18
```
Comment (rheacangeo) [inline tests/test_translate.py:23]: I think I'd be ok with changing the near0 for all of them to 1e-15. 1e-18 was somewhat arbitrary (was based on the differences I was seeing with the numpy backend), while we can reason about 1e-15 being at the floating point limit. But, it also makes sense we'd have different thresholds for GPU, so adding this capability is good.

File: tests/translate/translate_remap_profile_2d.py [hunk 5]
```diff
@@ -24,6 +24,7 @@ def __init__(self, grid):
             "a4_3": {"serialname": "q4_3", "istart": 0, "iend": grid.ie - 2},
             "a4_4": {"serialname": "q4_4", "istart": 0, "iend": grid.ie - 2},
         }
+        self.ignore_near_zero_errors = {"q4_4": True}
```
Comment (rheacangeo) [inline tests/translate/translate_remap_profile_2d.py:27]: interesting that it's just one of them

**Review by rheacangeo — APPROVED** [review]
You haven't asked for a review yet, so I'm sorry for jumping the gun. But this looks like good stuff!

File: fv3core/stencils/a2b_ord4.py (hunk 1 above)
Comment (eddie-c-davis) [reply fv3core/stencils/a2b_ord4.py:293]: Thank you, was just trying to simplify the code.

File: fv3core/stencils/remap_profile.py (hunk 2 above)
Comment (eddie-c-davis) [reply fv3core/stencils/remap_profile.py:301]: Yes, the conditionals have come a long way!

File: tests/translate/translate_remap_profile_2d.py (hunk 5 above)
Comment (eddie-c-davis) [reply tests/translate/translate_remap_profile_2d.py:27]: Agreed!

**Comment by eddie-c-davis** [comment]
Thank you for the preemptive review! There will be more GPU fixes to come, but I wanted to get these ones in before they get lost in the shuffle.

**Pull request merged**